src/vellum/lib/test_suites/resources.py

```diff
@@ -0,0 +1,184 @@
+from __future__ import annotations
+import os
+import time
+from typing import Callable
+
+from src.vellum.lib.test_suites.constants import DEFAULT_MAX_POLLING_DURATION_MS, DEFAULT_POLLING_INTERVAL_MS
+from src.vellum.lib.test_suites.exceptions import TestSuiteRunResultsException
+from src.vellum.types.named_test_case_variable_value_request import NamedTestCaseVariableValueRequest
+from src.vellum.types.test_case_variable_value import TestCaseVariableValue
+from src.vellum.client import Vellum
+from src.vellum.types import (
+    TestSuiteRunExecution,
+    TestSuiteRunMetricOutput,
+    TestSuiteRunState,
+    TestSuiteRunExternalExecConfigRequest,
+    TestSuiteRunExternalExecConfigDataRequest,
+    ExternalTestCaseExecution,
+)
+from src.vellum.lib.utils.paginator import PaginatedResults, get_all_results
+
+
+class VellumTestSuiteRunExecution(TestSuiteRunExecution):
+    @staticmethod
+    def from_api(execution: TestSuiteRunExecution) -> VellumTestSuiteRunExecution:
+        return VellumTestSuiteRunExecution(
+            id=execution.id,
+            test_case_id=execution.test_case_id,
+            outputs=execution.outputs,
+            metric_results=execution.metric_results,
+        )
+
+    def get_metric_output(
+        self,
+        metric_identifier: str | None = None,
+        output_identifier: str | None = None,
+    ) -> TestSuiteRunMetricOutput:
+        metric_outputs = self.get_metric_outputs(metric_identifier)
+
+        filtered_metric_outputs = [
+            metric_output
+            for metric_output in metric_outputs
+            if not output_identifier or metric_output.name == output_identifier
```

Comment:
nit: Avoids edge case of falsy values
```suggestion
            if output_identifier is None or metric_output.name == output_identifier
```